crates/api_common/src/utils.rs

```diff
@@ -592,3 +594,29 @@ pub async fn delete_user_account(person_id: PersonId, pool: &DbPool) -> Result<(
 
   Ok(())
 }
+
+pub async fn listing_type_with_site_default(
+  listing_type: Option<ListingType>,
+  pool: &DbPool,
+) -> Result<ListingType, LemmyError> {
+  Ok(match listing_type {
+    Some(l) => l,
+    None => {
+      let site = blocking(pool, Site::read_local_site).await??;
+      ListingType::from_str(&site.default_post_listing_type)?
+    }
+  })
+}
+
+/// Check to make sure a community_id or community_name is given for community listings
+pub fn check_missing_community_id(
+  listing_type: ListingType,
+  community_name: &Option<String>,
+  community_id: Option<CommunityId>,
+) -> Result<(), LemmyError> {
+  if listing_type == ListingType::Community && community_name.is_none() && community_id.is_none() {
+    Err(LemmyError::from_message("no_id_given"))
+  } else {
+    Ok(())
+  }
+}
```

Comment:
A much cleaner solution for the community name/id would be an enum that has both variants, and then play around with serde attributes to make it parse correctly. There should be some way to make it work, and then it wont be necessary to do any manual checks (invalid json will directly fail during parsing).

https://serde.rs/enum-representations.html

Reply:
We can do that later, as it would require an API change there. Also I'm not sure how that would work with JSON.

Reply:
I feel like there should be a way to get it working without changing the api, but not sure. I would add a test case that parses variations of valid json data, and then play around with the struct/enum definition and serde attrs until it works.

Reply:
JSON can't do enums or union types, it can only do numbers, strings, etc.

Reply:
Of course, but serde can.

Reply:
Okay it would probably look like this, i dont think there is a way to avoid the breaking change. Still i think its much cleaner to implement this way later.

```

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GetPosts {
  ...
  pub community: CommunityIdentifier,
  ...
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub enum CommunityIdentifier {
  Id(i32),
  Name(String)
}
```